orangecontrib/text/widgets/owscraper.py

```diff
@@ -0,0 +1,160 @@
+        self.author_chbox = gui.checkBox(self.text_includes_box, self,
+                                                 "includes_author",
+                                                 "Author")
+        self.date_chbox = gui.checkBox(self.text_includes_box, self,
+                                          "includes_date", "Date")
+        self.title_chbox = gui.checkBox(self.text_includes_box, self,
+                                           "includes_title", "Title")
+        self.web_url_chbox = gui.checkBox(self.text_includes_box, self,
+                                          "includes_web_url", "URL")
+
+        self.scraper_controls.append(self.article_chbox)
+        self.scraper_controls.append(self.author_chbox)
+        self.scraper_controls.append(self.date_chbox)
+        self.scraper_controls.append(self.title_chbox)
+        self.scraper_controls.append(self.web_url_chbox)
+
+
+        self.run_query_button = gui.button(self.controlArea, self, 'Run Query', tooltip="Run the chosen article Query")
+        self.run_query_button.setFocusPolicy(QtCore.Qt.NoFocus)
+        self.scraper_controls.append(self.run_query_button)
+
+
+    def date_to_str(input_date):
+        """
+        Returns a string representation of the input 
+        :param input_date:
+        :type input_date: datetime
+        :return: str
+        """
+        input_date = str(input_date.year) + str(input_date.month) + str(input_date.day) 
+        if input_date:
+            return input_date
+        else:
+            return"_"
+
+    def get_info(self, url):
+        """
+        Gets relevent available information from the url
+        :param url: 
+        :type url: string
+        :return : list
+        """
+        url_to_clean=url[4:-1] # remove irrevelent variables
+        article = Article(url_to_clean)
+        article.download()
+        article.parse()
+        try:
+            html_string = ElementTree.tostring(article.clean_top_node)
+        except:
+            html_string="_"
+        article.nlp()
+        scraped_data=[]
+        self.text = article.text if not article.text else "_"
+        self.title = article.title if not article.title else "_"
+        self.authors = str(', '.join(article.authors))   
+        self.authors= self.authors if  self.authors else "_"
+        self.date=_date_to_str(article.publish_date)
+        self.web_url=html_string if not html_string else "_"
+        scraped_data=[self.text, self.title, self.authors, self.date, self.web_url]
+        return scraped_data
+
+
+    def generate_table(self, data):
+
+
+        new_table=None
+                
+        text_includes_params = [self.includes_article, self.includes_title, self.includes_author,
+                                self.includes_date, self.includes_web_url]
+        required_text_fields = [incl_field for yes, incl_field in zip(text_includes_params, ARTICLE_TEXT_FIELDS) if yes]
```

Comment:
How so that `text_includes_params ` has only five elements when `ARTICLE_TEXT_FIELDS ` contains six? Also, the order of names and flags does not match.